Gateway/Http/Client/ApiRequestor.php

```diff
@@ -46,78 +48,69 @@ public static function post($url, $server_key, $data_hash)
-            } else {
-                $mergedHeders = [];
+        $curl = new Curl();
+        if (!$server_key) {
+            throw new Exception(
+                'The ServerKey/ClientKey is null, You need to set the server-key from Config. Please double-check Config and ServerKey key. ' .
+                'You can check from the Midtrans Dashboard. ' .
+                'See https://docs.midtrans.com/en/midtrans-account/overview?id=retrieving-api-access-keys ' .
+                'for the details.'
+            );
+        } else {
+            if ($server_key == "") {
+                throw new Exception(
+                    'The ServerKey/ClientKey is invalid, as it is an empty string. Please double-check your ServerKey key. ' .
+                    'You can check from the Midtrans Dashboard. ' .
+                    'See https://docs.midtrans.com/en/midtrans-account/overview?id=retrieving-api-access-keys ' .
+                    'for the details.'
+                );
             }
-
-            $curl_options = array_replace_recursive($curl_options, Config::$curlOptions, $headerOptions);
         }
 
-        if ($post) {
-            $curl_options[CURLOPT_POST] = 1;
-
-            if ($data_hash) {
-                $body = json_encode($data_hash);
-                $curl_options[CURLOPT_POSTFIELDS] = $body;
-            } else {
-                $curl_options[CURLOPT_POSTFIELDS] = '';
-            }
+        $headers = array(
+            "Content-Type" => "application/json",
+            "Accept" => "application/json",
+            "User-Agent" => "Magento 2 Module",
+            "X-Plugin-Name" => "midtrans-magento2-v2.6.5"
+        );
+        $curl->setOption(CURLOPT_RETURNTRANSFER, true);
+
+        // Set append notification to header
+        if (Config::$appendNotifUrl) $headers['X-Append-Notification'] = Config::$appendNotifUrl;
+        // Set override notification to header
+        if (Config::$overrideNotifUrl) $headers['X-Override-Notification'] = Config::$overrideNotifUrl;
+
+        $curl->setHeaders($headers);
+        $curl->setCredentials($server_key, "");
+
+        if ($method === "POST") {
+            $body = json_encode($data_hash);
+            $curl->post($url, $body);
+        } elseif ($method === "GET") {
+            $curl->get($url);
         }
 
-        curl_setopt_array($ch, $curl_options);
-
-        // For testing purpose
-        if (class_exists('\Midtrans\VT_Tests') && VT_Tests::$stubHttp) {
-            $result = self::processStubed($curl_options, $url, $server_key, $data_hash, $post);
-        } else {
-            $result = curl_exec($ch);
-            // curl_close($ch);
-        }
+        $result = $curl->getBody();
 
-        if ($result === false) {
-            throw new \Exception('CURL Error: ' . curl_error($ch), curl_errno($ch));
+        if ($result === '') {
+            $curl->doError("Midtrans Error: Something went wrong when request with Magento CURL");
```

Comment:
I don't think we need to check $result manually anymore, if mage $curl request encounter error, [it will auto throw exception](https://github.com/pepe1518/magento2/blob/89adbabc2a295c9b6f88768d541fd2858f59aae5/vendor/magento/framework/HTTP/Client/Curl.php#L388-L390).

Or if you want to throw custom exception, put try-catch around `$curl->get/post` instead.

Reply:
@rizdaprasetya I agree with that da. Removed the result check yaa